docs/RemoteCactbot.md

```diff
@@ -0,0 +1,271 @@
+See: [https://www.howtogeek.com/66214/how-to-forward-ports-on-your-router/](https://www.howtogeek.com/66214/how-to-forward-ports-on-your-router/).
+This is complicated and router-specific, and also gives you less control.
+This guide will not cover this as an option,
+but this may be more comfortable for you if you are network savvy.
+
+### Sign Up For ngrok
+
+First, sign up for [ngrok](https://ngrok.com/), if you don't have an account:
+<https://dashboard.ngrok.com/signup>
+
+The free account will cover everything you need to do.
+The worst part about a free account is that the url will change every time you start ngrok,
+and you will need to re-share this with everybody.
+
+There are alternate services you can use to do this sort of tunneling,
+but this guide will not cover them.
+
+### Download And Install
+
+Once you've logged in, click the "Download for Windows" button.
+This will give you a zip file.
+Place the `ngrok.exe` file somewhere in your path.
+
+The rest of this section assumes you're not that comfy with a command line.
+
+In this example, I'll unzip `ngrok.exe` directly into my `C:\Users\tinipoutini\` folder.
+Putting this in your own user folder or any other place in your `PATH` is a good choice.
+
+Now, open up `cmd.exe` by hitting Windows Key+R,
+then typing cmd, and hitting enter.
+This will open up a command line that looks something like this:
+
+```Batchfile
+Microsoft Windows [Version 6.1.7601]
+Copyright (c) 2009 Microsoft Corporation.  All rights reserved.
+
+C:\Users\tinipoutini>
+```
+
+The first thing to do is to set up your auth token.
+The command to do this is listed in your [ngrok dashboard](https://dashboard.ngrok.com/get-started/setup).
+The below is an example authtoken and will not work.
+Replace it with your own.
+
+```Batchfile
+C:\Users\tinipoutini>ngrok authtoken FKdtYBn57YTfpTRcqhr80xD_bW2KscKKIMRiHvIcDWeeZpEak
+Authtoken saved to configuration file: C:\Users\tinipoutini/.ngrok2/ngrok.yml
+```
+
+Next, start up the ngrok server via a command like this:
+
+```Batchfile
+ngrok http -host-header=rewrite -inspect=false --bind-tls "true" https://localhost/10501
+```
+
+This will create a window with server information.
+Closing this window or hitting Ctrl+C will close the server.
+
+```text
+ngrok by @inconshreveable                                       (Ctrl+C to quit)
+
+Session Status                online
+Account                       tini poutini (Plan: Free)
+Version                       2.3.35
+Region                        United States (us)
+Web Interface                 http://127.0.0.1:4040
+Forwarding                    https://a31a5403.ngrok.io -> https://localhost:105
+
+Connections                   ttl     opn     rt1     rt5     p50     p90
+                              0       0       0.00    0.00    0.00    0.00
+```
+
+You will need to run this last step any time you want to start a server.
```

Comment:
I saw your demo a while back of built-in ngrok support in OverlayPlugin.  Are you planning to merge that in any time soon? I was holding off on landing this pull request because almost all of it is moot with your changes, but if it's going to be a while before that gets merged, then I'll clean this up.